### PR TITLE
Add support for re-introduced warnings in scala 3.3.0

### PR DESCRIPTION
**Pull request by amumurst**
This PR prepares the plugin for the official scala 3.3.0 release (it is tagged, just not released quite yet https://github.com/lampepfl/dotty/releases/tag/3.3.0). It adds back all the warnings for unused and discarded values that is re-added in this release.

### CODE REVIEW

**Comment by amumurst** [comment]
Note that this adds the specific `unused:X` flags and not the `unused:all`. I have no strong opinion on which is better for the plugin. Adding `all` would mean that if another is added in 3.3.1, users of the plugin would have it automatically added when upgrading the scala version in their project.

**Comment by KamilPostrozny** [comment]
@amumurst personally I would go with `unused:all` for reasons you have mentioned. Note that for now all scala versions that support any `unused` option also support the rest of them. So for now it doesn't make sense - at least for me - to separate them.

**Review by lukestephenson-zendesk — APPROVED** [review]

File: plugin/src/main/scala/io/github/davidgregory084/ScalaVersion.scala [hunk 1]
```diff
@@ -21,6 +21,8 @@ import scala.Ordering.Implicits._
 case class ScalaVersion(major: Long, minor: Long, patch: Long) {
   def isBetween(addedVersion: ScalaVersion, removedVersion: ScalaVersion) =
     this >= addedVersion && this < removedVersion
+
+  def isAfter(addedVersion: ScalaVersion) = this >= addedVersion
```
Comment (armanbilge) [inline plugin/src/main/scala/io/github/davidgregory084/ScalaVersion.scala:25]: Since it's non-strict? just bikeshed

```suggestion
  def isOrAfter(addedVersion: ScalaVersion) = this >= addedVersion
```

File: plugin/src/sbt-test/sbt-tpolecat/scalacOptions/build.sbt [hunk 2]
```diff
@@ -185,6 +187,27 @@ val Scala31Options =
...
+    "-Ykind-projector",
+    "-Wvalue-discard",
+    "-Wunused:implicits",
+    "-Wunused:explicits",
+    "-Wunused:imports",
+    "-Wunused:locals",
+    "-Wunused:params",
+    "-Wunused:privates",
+    "-source",
+    "3.0-migration",
```
Comment (armanbilge) [inline plugin/src/sbt-test/sbt-tpolecat/scalacOptions/build.sbt:209]: sbt-typelevel was making this mistake as well. It seems this is not the sort of thing a plugin should be doing. But that will have to be fixed in another PR / breaking release.

> Without the project author explicitly opting into it. I think -source is dangerous and should not be enabled without thinking through all the consequences (and ideally it should only be used temporarily during a migration)

- https://github.com/lampepfl/dotty/discussions/17591#discussioncomment-6003475

**Review by armanbilge — APPROVED** [review]
We've enabled these same flags in the sbt-typelevel 0.5.0 RCs and they seem to be working well. :shipit:

If a regular maintainer neither objects nor steps up I will try and get this out later this week. Sorry for the delay folks 😅

**Comment by armanbilge** [comment]
I can merge this PR, but publishing appears to be broken 😕 if the credentials are borked which is what I suspect, I don't think I can personally fix that.

One path forward would be to start publishing under `org.typelevel` since we have org-wide credentials. That may require re-working the build and CI a bit: either to make our credential setup work with the existing sbt-ci-release, or switching to sbt-typelevel(-ci-release). Unfortunately I don't have time to look into this: I think we'll need someone new to step up as a regular maintainer.

File: plugin/src/main/scala/io/github/davidgregory084/ScalaVersion.scala (hunk 1 above)
Comment (cb372) [reply plugin/src/main/scala/io/github/davidgregory084/ScalaVersion.scala:25]: `isAtLeast`?

**Comment by DavidGregory084** [comment]
Apologies all, I'm really struggling to make time for sbt-tpolecat - life and work are conspiring against me atm. This PR looks absolutely great though, and I think we should merge it!

However, I'm not sure I can fix publishing either as I don't have access to repository level secrets in this repo anymore:
image

I think @armanbilge is right that this means publishing under `org.typelevel`, or alternatively, we can find a way to get the previous creds handed over to one of the org admins until the repo is in the right place to publish under `org.typelevel`?

On the plus side, there is an [artifact migration](https://github.com/scala-steward-org/scala-steward/pull/2587/files) ready to go in Scala Steward if we decide to go with the former route.

**Comment by armanbilge** [comment]
@typelevel/steering would one of you be able to give @DavidGregory084 admin access to this repo so they can install updated secrets? Thanks!

**Comment by DavidGregory084** [comment]
Thanks @samspills @armanbilge!

Publishing creds should be fixed up now: although I no longer use LastPass, I changed my Sonatype password as a precaution after the LastPass breach.

**Comment by DavidGregory084** [comment]
Thanks @amumurst!

File: plugin/src/sbt-test/sbt-tpolecat/scalacOptions/build.sbt (hunk 2 above)
Comment (DavidGregory084) [reply plugin/src/sbt-test/sbt-tpolecat/scalacOptions/build.sbt:209]: Don't worry - this isn't actually enabled by default, it's just enabled in the test suite to make sure that users can manipulate options for each mode separately.

Comment (DavidGregory084) [reply plugin/src/sbt-test/sbt-tpolecat/scalacOptions/build.sbt:209]: (see https://github.com/typelevel/sbt-tpolecat/pull/147/files#diff-25d6f7f6dbab5ef8591d0bce4e831ab4f799818e2d946938635f0f87c0a489afR25)
